koap-core/src/commonMain/kotlin/Message.kt

```diff
@@ -387,6 +407,112 @@ sealed class Message {
             }
         }
 
+        interface Block {
+            val number: Int
+            val more: Boolean
+            val size: Size
+
+            @Suppress("ktlint:standard:enum-entry-name-case")
+            enum class Size(internal val intValue: Int) {
+                `16`(0),
+                `32`(1),
+                `64`(2),
+                `128`(3),
+                `256`(4),
+                `512`(5),
+                `1024`(6),
```

Comment:
I was torn on the best approach here: I don't like the ticks but prefixes felt ugly as well.
Can't decide on which is the lesser of the evils. 🙈

Reply:
could go for `kB` for 1024, and probably cover 90+% of use cases! 😉

Reply:
I think Size should have an easy way to access the actual block size in bytes as a number.
For BERT blocks, the basic block size stays 1kB, just that you can have more in one message.

Wouldn't `szxValue` be more clear than `intValue` (at least if you look it up in the rfc)?

Reply:
using ordinal to match the szx value, they would be free to hold a single public size: Int value

Reply:
> Wouldn't `szxValue` be more clear than `intValue` (at least if you look it up in the rfc)?

I agree that w.r.t. the RFC it is much clearer, but I would like KoAP to be a little more intuitive/approachable (whereas it doesn't require users to look at the RFCs). At least, that is the hope.

Reply:
> I think Size should have an easy way to access the actual block size in bytes as a number.

I've exposed the value as `size` property on `Size`. A bit redundant; assuming block sizes are always in bytes, perhaps it would make sense to name the property `bytes`?

Reply:
> I was torn on the best approach here: I don't like the ticks but prefixes felt ugly as well. Can't decide on which is the lesser of the evils. 🙈

I ended up adding a `blockSizeOf(int, ..)` function that can be optionally used and makes for nicer syntax. Not sure if I'm adding unnecessary complexity with it though. 🤷

Reply:
> > I think Size should have an easy way to access the actual block size in bytes as a number.
> 
> I've exposed the value as `size` property on `Size`. A bit redundant; assuming block sizes are always in bytes, perhaps it would make sense to name the property `bytes`?

hm. maybe the field and type should be called `blockSize`/`BlockSize` rather than just `size`/`Size`? then I think `bytes` makes sense, and is the same as in `Size1` and `Size2`.